src/firefox/index.js

```diff
@@ -249,9 +250,51 @@ export async function useProfile(
     app,
     configureThisProfile = configureProfile,
     customPrefs = {},
+    searchProfilesPath = '',
   }: UseProfileParams = {},
 ): Promise<FirefoxProfile> {
-  const profile = new FirefoxProfile({destinationDirectory: profilePath});
+  let profile;
+  const finder = new FirefoxProfile.Finder(searchProfilesPath);
+  const finderGetPath = promisify(finder.getPath, finder);
+  const hasProfileName = (profileName) => {
+    return finder.profiles.filter(
+      (profileDef) => profileDef.Name === profileName
+    ).length !== 0;
+  };
+
+  try {
+    const dirExists = await isDirectory(profilePath);
+    const defaultProfilePath = (
+      hasProfileName('default') && await finderGetPath('default')
+    );
+    const defaultDevProfilePath = (
+      hasProfileName('dev-edition-default') &&
+        await finderGetPath('dev-edition-default')
+    );
+    if (dirExists) {
+      log.debug(`Copying profile directory from "${profilePath}"`);
+      if (profilePath === defaultProfilePath ||
+        profilePath === defaultDevProfilePath) {
+        throw new UsageError(
+          `Cannot use profile at "${profilePath}"`
+        );
+      }
+      profile = new FirefoxProfile({destinationDirectory: profilePath});
+    } else {
+      log.debug(`Assuming ${profilePath} is a named profile`);
+      const profileDirectory = await finderGetPath(profilePath);
+      if (profileDirectory === defaultProfilePath ||
+            profileDirectory === defaultDevProfilePath) {
+        throw new UsageError(
+          `Cannot use named profile "${profilePath}"`
+        );
+      }
+      profile = new FirefoxProfile({destinationDirectory: profileDirectory});
+    }
+  } catch (error) {
+    throw new WebExtError(
+      `Could not copy Firefox profile from ${profilePath}: ${error}`);
```

Comment:
As the message logged at line 275, this error message is also not the right one.